src/main/kotlin/calculator/StringCalculator.kt

```diff
@@ -0,0 +1,61 @@
+package calculator
+
+class StringCalculator(private val str: String?) {
+
+    fun calculate(): Int = when (str.isNullOrBlank()) {
+        true -> { RESULT_EMPTY_VALUE }
+        false -> {
+            val matchResult: MatchResult? = Regex(REGEX_DIVIDER).find(str)
+            val divider = getDivider(matchResult)
+
+            require(!divider.isNullOrEmpty()) { "divider" }
+
+            val splitStr = splitStr(divider, matchResult)
+            val numbers = splitToNumbers(splitStr)
+            add(numbers)
+        }
+    }
+
+    private fun getDivider(matchResult: MatchResult?): String = when (matchResult) {
+        null -> DEFAULT_DIVIDER
+        else -> getCustomDivider(matchResult)
+    }
+
+    private fun getCustomDivider(matchResult: MatchResult): String =
+        matchResult.let {
+            it.groupValues[1]
+        }
+
+    private fun splitStr(divider: String, matchResult: MatchResult?): List<String> {
+        val convertString: String = when (matchResult) {
+            null -> str!!
+            else -> matchResult.let { it.groupValues[2] }
+        }
+
+        return convertString.split(divider.toRegex())
+    }
+
+    private fun splitToNumbers(splitStr: List<String>): List<Int> {
+        val result = ArrayList<Int>()
+        splitStr.forEach { str ->
+            val number: UInt = str.toUInt()
+            result.add(number.toInt())
+        }
```

Comment:
map API 를 활용하면 아래와 같이 구현할수 있을것 같습니다 ㅎㅎ 
```suggestion
        splitStr.map { number.toInt() }
```
추가로 현재는 양수를 보장할수 없는 구조라고 생각해요. 
양수를 보장하는 객체를 생성하여 사용해보세요 😄 
이번 과정을 진행하며 객체를 작게 나누는것에 대해 경험해보는것을 권장 드립니다 !